src/Forms/GridField/GridState_Data.php

```diff
@@ -16,23 +16,42 @@ class GridState_Data
      */
     protected $data;
 
+    protected $defaults = [];
+
     public function __construct($data = [])
     {
         $this->data = $data;
     }
 
     public function __get($name)
     {
-        return $this->getData($name, new GridState_Data());
+        return $this->getData($name, new self());
     }
 
     public function __call($name, $arguments)
     {
         // Assume first parameter is default value
-        $default = empty($arguments) ? new GridState_Data() : $arguments[0];
+        if (empty($arguments)) {
+            $default = new self();
+        } else {
+            $default = $arguments[0];
+        }
+
         return $this->getData($name, $default);
     }
 
+    /**
+     * Initialise the defaults values for the grid field state
+     * @param array $defaults
+     */
+    public function initDefaults(array $defaults): void
+    {
+        foreach ($defaults as $key => $name) {
+            $this->defaults[$key] = $name;
```

Comment:
Shouldn't getData's 2nd argument populate the defaults property? It would also render the initDefaults method optional as you could aways simply use getData.

Reply:
OK I've thought about this more and I suspect that this argument gets used in grid field extensions as the setter. This means that if no default was specified we would risk data loss. Better to risk unnecessarily verbose URLs than breakages, so forcing the explicit use of initDefaults(), as you originally developed, makes sense.